### PR TITLE
Add missing specializations in execution policies

### PR DESCRIPTION
**Pull request by mcopik**
Proper support for rebounded policies.

### CODE REVIEW

File: hpx/parallel/execution_policy.hpp [hunk 1]
```diff
@@ -1418,6 +1423,11 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
         template <>
         struct is_parallel_execution_policy<parallel_task_execution_policy>
           : std::true_type
+        {};        
+
+        template <typename Executor, typename Parameters>
+        struct is_parallel_execution_policy<parallel_task_execution_policy_shim<Executor, Parameters>>
+          : std::true_type
         {};
         /// \endcond
```
Comment (hkaiser) [inline hpx/parallel/execution_policy.hpp]: The `parallel_task_...` policies are 'parallel' as well. Could you add those while you're at it as well, please?

File: hpx/parallel/execution_policy.hpp [hunk 2]
```diff
@@ -1508,10 +1518,22 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
           : std::true_type
         {};
 
+        template <typename Executor, typename Parameters>
+        struct is_async_execution_policy<
+                sequential_execution_policy_shim<Executor, Parameters> >
+          : std::true_type
+        {};
```
Comment (hkaiser) [inline hpx/parallel/execution_policy.hpp]: You probably wanted to add the `parallel_task_execution_policy_shim<>` instead.

File: hpx/parallel/execution_policy.hpp [hunk 3]
```diff
@@ -1508,10 +1518,22 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
...
         template <>
         struct is_async_execution_policy<parallel_task_execution_policy>
           : std::true_type
         {};
+
+        template <typename Executor, typename Parameters>
+        struct is_async_execution_policy<
+                parallel_execution_policy_shim<Executor, Parameters> >
+          : std::true_type
+        {};
```
Comment (hkaiser) [inline hpx/parallel/execution_policy.hpp]: Same here, should read `parallel_task_execution_policy_shim<>` instead.

File: hpx/parallel/execution_policy.hpp (hunk 1 above)
Comment (mcopik) [inline hpx/parallel/execution_policy.hpp]: Now 'parallel' are parallel, parallel_shim, parallel_task, parallel_task_shim and parallel_vector.
Which other am I missing in is_parallel_execution_policy? Or did you mean something else by 'add'?

File: hpx/parallel/execution_policy.hpp (hunk 2 above)
Comment (mcopik) [inline hpx/parallel/execution_policy.hpp]: My mistake, fixed!

File: hpx/parallel/execution_policy.hpp (hunk 3 above)
Comment (mcopik) [inline hpx/parallel/execution_policy.hpp]: My mistake, fixed!

**Comment by hkaiser** [comment]
Inspect is not happy with your changes: https://circle-artifacts.com/gh/STEllAR-GROUP/hpx/3716/artifacts/0/tmp/circle-artifacts.87XsJ13/hpx_inspect_report.html

File: hpx/parallel/execution_policy.hpp (hunk 1 above)
Comment (hkaiser) [inline hpx/parallel/execution_policy.hpp]: That's what I meant, thanks!

Comment (mcopik) [inline hpx/parallel/execution_policy.hpp]: It was already there, I think the diff was misleading ;-)

**Pull request merged**